scripts/feature_tests/upgrade/upgrade_vars.sh

```diff
@@ -1,13 +1,19 @@
 #!/bin/bash
+set -eux
 
 # Folder created for specific capi release when running
 # ${CLUSTER_API_REPO}/cmd/clusterctl/hack/create-local-repository.py
 # For CAPI release v0.3.12, the folder v0.3.8 is created
 
 export CAPIRELEASE_HARDCODED="v0.3.8"
 
+function get_capm3_latest() {
+    clusterctl upgrade plan | grep infrastructure-metal3 | awk 'NR == 1 {print $5}'
+}
+
 export CAPM3RELEASE="v0.4.0"
-export CAPM3_REL_TO_VERSION="v0.4.1"
+CAPM3_REL_TO_VERSION="$(get_capm3_latest)" || true
+export CAPM3_REL_TO_VERSION
 
 export CAPIRELEASE="v0.3.15"
 export CAPI_REL_TO_VERSION="v0.3.16"
```

Comment:
These could be bumped/automated as well. The latest release is v0.3.17.

Reply:
+1 for automating them as well

Reply:
Agree, however, to my surprise, v0.3.17 is not suggested as latest version by clusterctl?

```
`$ clusterctl upgrade plan
Checking cert-manager version...
Cert-Manager will be upgraded from "v0.11.0" to "v1.1.0"

Checking new release availability...

Management group: capi-system/cluster-api, latest release available for the v1alpha3 API Version of Cluster API (contract):

NAME                    NAMESPACE                           TYPE                     CURRENT VERSION   NEXT VERSION
bootstrap-kubeadm       capi-kubeadm-bootstrap-system       BootstrapProvider        v0.3.16           Already up to date
control-plane-kubeadm   capi-kubeadm-control-plane-system   ControlPlaneProvider     v0.3.16           Already up to date
cluster-api             capi-system                         CoreProvider             v0.3.16           Already up to date
infrastructure-metal3   capm3-system                        InfrastructureProvider   v0.4.2            Already up to date

You are already up to date!`
```

Reply:
That's strange. :open_mouth: For me it suggests the latest version (otherwise I would probably have missed it :smile: )

```console
$ clusterctl upgrade plan
Checking cert-manager version...
Cert-Manager will be upgraded from "v0.11.0" to "v1.1.0"

Checking new release availability...

Management group: capi-system/cluster-api, latest release available for the v1alpha3 API Version of Cluster API (contract):

NAME                    NAMESPACE                           TYPE                     CURRENT VERSION   NEXT VERSION
bootstrap-kubeadm       capi-kubeadm-bootstrap-system       BootstrapProvider        v0.3.16           Already up to date
control-plane-kubeadm   capi-kubeadm-control-plane-system   ControlPlaneProvider     v0.3.16           Already up to date
cluster-api             capi-system                         CoreProvider             v0.3.16           Already up to date
infrastructure-metal3   capm3-system                        InfrastructureProvider   v0.4.2            Already up to date

You are already up to date!


New clusterctl version available: v0.3.15 -> v0.3.17
https://github.com/kubernetes-sigs/cluster-api/releases/tag/v0.3.17
```

Reply:
Could it be that you already have the latest for some reason? What does `clusterctl version` say?

Reply:
I think it is because of hardcoded version of `CAPI_REL_TO_VERSION` we have.